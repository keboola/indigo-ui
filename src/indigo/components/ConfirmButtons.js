import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheckCircle } from '@fortawesome/free-solid-svg-icons/faCheckCircle';
import { Button, ButtonToolbar } from 'react-bootstrap';
import Loader from './Loader';

class ConfirmButtons extends React.Component {
  render() {
    if (this.props.block) {
      return this.renderSaveButton();
    }

    return (
      <ButtonToolbar className={classnames('btn-toolbar-confirm', this.props.className)}>
        {this.props.isSaving && <Loader />}
        {this.props.children}
        {!this.props.showCancel && ' '}
        {this.renderCancelButton()}
        {this.renderSaveButton()}
      </ButtonToolbar>
    );
  }

  renderSaveButton() {
    if (!this.props.showSave) {
      return null;
    }

    return (
      <Button
        className="btn-confirm"
        block={this.props.block}
        type={this.props.saveButtonType}
        bsStyle={this.props.saveStyle}
        disabled={this.props.isSaving || this.props.isDisabled}
        onClick={this.props.onSave}
      >
        {this.props.block && (
          <>
            {this.props.isSaving ? (
              <Loader className="icon-addon-right" />
            ) : (
              this.props.saveStyle !== 'danger' && (
                <FontAwesomeIcon icon={faCheckCircle} className="icon-addon-right" />
              )
            )}
          </>
        )}
        {this.props.saveLabel}
      </Button>
    );
  }

  renderCancelButton() {
    if (!this.props.showCancel) {
      return null;
    }

    return (
      <Button
        className="btn-confirm"
        bsStyle="link"
        disabled={this.props.isSaving}
        onClick={this.props.onCancel}
      >
        {this.props.cancelLabel}
      </Button>
    );
  }
}

ConfirmButtons.defaultProps = {
  onSave: () => {},
  onCancel: () => {},
  saveLabel: 'Save',
  saveStyle: 'success',
  cancelLabel: 'Cancel',
  saveButtonType: 'button',
  isSaving: false,
  isDisabled: false,
  showSave: true,
  showCancel: true,
  block: false,
};

ConfirmButtons.propTypes = {
  isSaving: PropTypes.bool,
  isDisabled: PropTypes.bool,
  cancelLabel: PropTypes.string,
  saveLabel: PropTypes.string,
  saveStyle: PropTypes.string,
  onCancel: PropTypes.func,
  onSave: PropTypes.func,
  saveButtonType: PropTypes.oneOf(['button', 'submit']),
  showCancel: PropTypes.bool,
  showSave: PropTypes.bool,
  block: PropTypes.bool,
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  children: PropTypes.any,
};

export default ConfirmButtons;
