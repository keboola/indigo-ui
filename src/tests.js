import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
