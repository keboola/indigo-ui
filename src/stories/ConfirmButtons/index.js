import React from 'react';
import { storiesOf } from '@storybook/react';
import { Button } from 'react-bootstrap';
import { action } from '@storybook/addon-actions';
import ConfirmButtons from '../../indigo/components/ConfirmButtons';

storiesOf('ConfirmButtons', module)
  .add('Basic Init', () => <ConfirmButtons />)
  .add('Saving', () => <ConfirmButtons isSaving />)
  .add('Disabled', () => <ConfirmButtons isDisabled />)
  .add('Show Save button', () => <ConfirmButtons showSave={false} />)
  .add('Show Cancel button', () => <ConfirmButtons showCancel={false} />)
  .add('Additional buttons', () => (
    <ConfirmButtons>
      <Button>Try Query</Button>
    </ConfirmButtons>
  ))
  .add('Cancel label', () => <ConfirmButtons cancelLabel="Reset" />)
  .add('Save label', () => <ConfirmButtons saveLabel="Save Configuration" />)
  .add('Save button style', () => <ConfirmButtons saveStyle="primary" />)
  .add('Save button type', () => <ConfirmButtons saveButtonType="submit" />)
  .add('Additional class', () => (
    <div className="clearfix">
      <ConfirmButtons className="pull-right" />
    </div>
  ))
  .add('Events demo', () => (
    <div className="clearfix">
      <ConfirmButtons onSave={action('event - save')} onCancel={action('event - cancel')} />
    </div>
  ));
