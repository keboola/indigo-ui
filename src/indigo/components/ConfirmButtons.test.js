import React from 'react';
import { Button } from 'react-bootstrap';
import { matchesSnapshot } from '../../tests';
import ConfirmButtons from './ConfirmButtons';

describe('<ConfirmButtons />', () => {
  it('Basic Init', () => {
    matchesSnapshot(<ConfirmButtons onSave={() => null} isSaving={false} />);
  });

  it('Saving', () => {
    matchesSnapshot(<ConfirmButtons onSave={() => null} isSaving />);
  });

  it('Disabled', () => {
    matchesSnapshot(<ConfirmButtons isDisabled onSave={() => null} isSaving={false} />);
  });

  it('Show Save button', () => {
    matchesSnapshot(<ConfirmButtons showSave={false} onSave={() => null} isSaving={false} />);
  });

  it('Show Cancel button', () => {
    matchesSnapshot(<ConfirmButtons showCancel={false} onSave={() => null} isSaving={false} />);
  });

  it('Additional buttons', () => {
    matchesSnapshot(
      <ConfirmButtons onSave={() => null} isSaving={false}>
        <Button>Try Query</Button>
      </ConfirmButtons>
    );
  });

  it('Cancel label', () => {
    matchesSnapshot(<ConfirmButtons cancelLabel="Reset" onSave={() => null} isSaving={false} />);
  });

  it('Save label', () => {
    matchesSnapshot(
      <ConfirmButtons saveLabel="Save Configuration" onSave={() => null} isSaving={false} />
    );
  });

  it('Save button style', () => {
    matchesSnapshot(<ConfirmButtons saveStyle="primary" onSave={() => null} isSaving={false} />);
  });

  it('Save button type', () => {
    matchesSnapshot(
      <ConfirmButtons saveButtonType="submit" onSave={() => null} isSaving={false} />
    );
  });

  it('Additional class', () => {
    matchesSnapshot(
      <div className="clearfix">
        <ConfirmButtons className="pull-right" onSave={() => null} isSaving={false} />
      </div>
    );
  });
});
