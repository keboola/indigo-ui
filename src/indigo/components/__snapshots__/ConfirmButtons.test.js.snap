// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Check /> Additional buttons 1`] = `
<div
  className="btn-toolbar-confirm btn-toolbar"
  role="toolbar"
>
  <button
    className="btn btn-default"
    disabled={false}
    type="button"
  >
    Try Query
  </button>
  <button
    className="btn-confirm btn btn-link"
    disabled={false}
    onClick={undefined}
    type="button"
  >
    Cancel
  </button>
  <button
    className="btn-confirm btn btn-success"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Save
  </button>
</div>
`;

exports[`<Check /> Additional class 1`] = `
<div
  className="clearfix"
>
  <div
    className="btn-toolbar-confirm pull-right btn-toolbar"
    role="toolbar"
  >
    <button
      className="btn-confirm btn btn-link"
      disabled={false}
      onClick={undefined}
      type="button"
    >
      Cancel
    </button>
    <button
      className="btn-confirm btn btn-success"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Save
    </button>
  </div>
</div>
`;

exports[`<Check /> Basic Init 1`] = `
<div
  className="btn-toolbar-confirm btn-toolbar"
  role="toolbar"
>
  <button
    className="btn-confirm btn btn-link"
    disabled={false}
    onClick={undefined}
    type="button"
  >
    Cancel
  </button>
  <button
    className="btn-confirm btn btn-success"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Save
  </button>
</div>
`;

exports[`<Check /> Cancel label 1`] = `
<div
  className="btn-toolbar-confirm btn-toolbar"
  role="toolbar"
>
  <button
    className="btn-confirm btn btn-link"
    disabled={false}
    onClick={undefined}
    type="button"
  >
    Reset
  </button>
  <button
    className="btn-confirm btn btn-success"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Save
  </button>
</div>
`;

exports[`<Check /> Disabled 1`] = `
<div
  className="btn-toolbar-confirm btn-toolbar"
  role="toolbar"
>
  <button
    className="btn-confirm btn btn-link"
    disabled={false}
    onClick={undefined}
    type="button"
  >
    Cancel
  </button>
  <button
    className="btn-confirm btn btn-success"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Save
  </button>
</div>
`;

exports[`<Check /> Save button style 1`] = `
<div
  className="btn-toolbar-confirm btn-toolbar"
  role="toolbar"
>
  <button
    className="btn-confirm btn btn-link"
    disabled={false}
    onClick={undefined}
    type="button"
  >
    Cancel
  </button>
  <button
    className="btn-confirm btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Save
  </button>
</div>
`;

exports[`<Check /> Save button type 1`] = `
<div
  className="btn-toolbar-confirm btn-toolbar"
  role="toolbar"
>
  <button
    className="btn-confirm btn btn-link"
    disabled={false}
    onClick={undefined}
    type="button"
  >
    Cancel
  </button>
  <button
    className="btn-confirm btn btn-success"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
    Save
  </button>
</div>
`;

exports[`<Check /> Save label 1`] = `
<div
  className="btn-toolbar-confirm btn-toolbar"
  role="toolbar"
>
  <button
    className="btn-confirm btn btn-link"
    disabled={false}
    onClick={undefined}
    type="button"
  >
    Cancel
  </button>
  <button
    className="btn-confirm btn btn-success"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Save Configuration
  </button>
</div>
`;

exports[`<Check /> Saving 1`] = `
<div
  className="btn-toolbar-confirm btn-toolbar"
  role="toolbar"
>
  <span
    className="fa fa-spin fa-spinner"
  />
  <button
    className="btn-confirm btn btn-link"
    disabled={true}
    onClick={undefined}
    type="button"
  >
    Cancel
  </button>
  <button
    className="btn-confirm btn btn-success"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Save
  </button>
</div>
`;

exports[`<Check /> Show Cancel button 1`] = `
<div
  className="btn-toolbar-confirm btn-toolbar"
  role="toolbar"
>
  <button
    className="btn-confirm btn btn-success"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Save
  </button>
</div>
`;

exports[`<Check /> Show Save button 1`] = `
<div
  className="btn-toolbar-confirm btn-toolbar"
  role="toolbar"
>
  <button
    className="btn-confirm btn btn-link"
    disabled={false}
    onClick={undefined}
    type="button"
  >
    Cancel
  </button>
</div>
`;
